docs/DeveloperGuide.md

```diff
@@ -1,34 +1,442 @@
+
+To initialise this component, the `SortedMap<String, Patient>` from the Data class is passed during object creation.
+This SortedMap<String, Patient> is the entire list of patients and their records.
+
+Then, we can call the `save(SortedMap<String, Patient> patientData)` method to save the SortedMap into a file whenever changes are made.
+The file is specified by the variable FILE_PATH in Constants class.
+
+The reverse process would be the `load()` method. The file from the specified FILE_PATH is loaded into a String, which is 
+converted to a SortedMap<String, Patient> afterwards.
+
+`Storage.java`:
+
+- facilitates the saving of application data into a text file
+- facilitates the loading of application data from the aforementioned text file
+- convert records to string
+- converts string to records
+
+### Exception Component
+
+API: `BaseException.java` and its subclasses
+
+`BaseException.java`:
+- handles all exceptions that occur during the execution of PatientManager
+- can report an error message, prompting the user to provide a syntatically correct command
+- may also report the cause of error for debugging purposes
+
+### Commons
+
+Constants used by multiple classes are stored in the `seedu.duke.Constants` class.
+
+## Implementation
+
+This section describes some noteworthy details on how certain details are implemented.
+
+### Tokenizing User Input
+
+The parser is one of the core components in charge of parsing all user input commands into program-understandable commands and
+arguments. For the ease of expansion of this program's functionality as well as for its testability, reflection is used to invoke
+commands.
+
+First is the initialization of this parser. A `Ui` instance and a `Data` instance is passed and stored. This is important as
+these two will be passed to logic components (command classes) later.
+
+Then, we can parse a user-input string by passing it to `parse()`. We use an example of this:
+```
+record 01/05/2021 /s coughing, fever /p panadol Paracetamol 500mg*20
+```
+
+This is broken into a few steps:
+
+1. Initialize an empty hashmap, called `arguments`.
+1. Tokenize using **any number of consecutive white spaces**.
+1. Taken out the first token as command, i.e. `record`. Push it into the hash map using **key** `command`.
+   Create a new empty **list** with default **key** `payload`.
+1. Check if next token starts with `/`. No, so we add it to the list: `list = ['payload']`.
+1. Check if next token starts with `/`. Yes, so we concatenate all tokens in the list to one string use delimiter ` `
+   (empty whitespace). Put it into the hash map using the key `payload`.
+   **Reset the list**, and set new key to `s` (the part after this `/`).
+1. Repeat same process, we have `list = ['coughing,']`
+1. Repeat same process, we have `list = ['coughing,', 'fever']`
+1. Same process, `coughing, fever` is pushed into arguments hash map with key `s`. Reset the list, and new key set to `p`.
+1. ... 
```

Comment:
This is a long list of lengthy steps, may want to include a sequence diagram to better visualise this?